vet-service/src/main/java/com/petclinic/vet/presentationlayer/VetController.java

```diff
@@ -76,6 +79,20 @@ public Mono<ResponseEntity<Double>> getAverageRatingByVetId(@PathVariable String
                 .map(ResponseEntity::ok)
                 .defaultIfEmpty(ResponseEntity.notFound().build());
     }
+    @GetMapping("{vetId}/ratings/date")
+    public Flux<RatingResponseDTO> getRatingsOfAVetBasedOnDate(@PathVariable String vetId, @RequestParam Map<String,String> queryParams) {
+        if (queryParams.containsKey("year")) {
+            String year = queryParams.get("year");
+
+            //This regex signifies that it required 4 numbers input for the year
+            if (!year.matches("^\\d{4}$")) {
+                throw new NotFoundException("Invalid year format. Please enter a valid year.");
+            }
+        }
```

Comment:
In a later bug, could handle if queryParams doesn't contain the key "year", but nevertheless, it's well-written code.